### DISCUSSION TITLE
Console Error When Running WASM

### DISCUSSION THREAD

**Discussion by evanshabsove** [bevyengine/bevy#6302] [Q&A] [answered]
Hi there!

I'm new to rust and to bevy so forgive me if this is a bit of a silly question.

I'm trying to get my game to run in the browser, I have followed the instructions on the bevy cheatbook [https://bevy-cheatbook.github.io/platforms/wasm.html](https://bevy-cheatbook.github.io/platforms/wasm.html)

But when I load up the wasm file using the command

```
wasm-server-runner target/wasm32-unknown-unknown/release/my_game.wasm
```

I get the following error in the console 

```
panicked at 'index out of bounds: the len is 6 but the index is 6', /Users/macbook/.cargo/registry/src/github.com-1ecc6299db9ec823/wgpu-hal-0.13.2/src/gles/queue.rs:358:25
```

I'm a bit lost in how to debug it all...just hoping for some direction 😄 any help is greatly appreciated! 

Cargo.toml file

```
[package]
name = "my_game"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
bevy = { version = "0.8.1", default-features = false, features = ["bevy_asset", "bevy_winit", "render", "png", "x11"] }
bevy_kira_audio = { version = "0.12", default-features = false, features = ["mp3"] }
bevy-inspector-egui = "0.12.1"
bevy_asset_loader = "0.12.1"
bevy_ecs_tilemap = "0.7.0"
bevy_rapier2d = "*"
bevy_ecs_ldtk = "0.4.0"

# Enable a small amount of optimization in debug mode
[profile.dev]
opt-level = 1

# Enable high optimizations for dependencies (incl. Bevy), but not for our code:
[profile.dev.package."*"]
opt-level = 3
```

wasm-bingen version

```
% cargo tree --invert wasm-bindgen --target wasm32-unknown-unknown --depth 0
> wasm-bindgen v0.2.83
```

**Comment by alice-i-cecile** [comment, marked as answer]
So, I'd go look in the wgpu code, and figure out what the line is. This is the tag for 0.13.2, so we'll start here: https://github.com/gfx-rs/wgpu/releases/tag/v0.13.2

Navigating, we find this file: https://github.com/gfx-rs/wgpu/blob/v0.13.2/wgpu-hal/src/gles/queue.rs

So the panic is happening on this line: https://github.com/gfx-rs/wgpu/blob/9f82504d8a3b6e234599b19cc35c299bcec8b138/wgpu-hal/src/gles/queue.rs#L358

Looks like... some sort of extraction process? And it's working with cubemaps? It's probably an off-by-one error.

This code was working fine on desktop, so it's almost certainly a bug. Report it upstream to `wgpu`, and work with them to find a minimal reproduction :)

**Reply by evanshabsove** [reply]
Great! I will report it to `wgpu` thanks for the help :)
